vespalib/src/vespa/vespalib/util/file_area_freelist.h

```diff
@@ -0,0 +1,27 @@
+// Copyright Verizon Media. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+
+#pragma once
+
+#include <cstdint>
+#include <limits>
+#include <map>
+#include <set>
+
+namespace vespalib::alloc {
+
+/*
+ * Class that tracks free areas in a file.
+ */
+class FileAreaFreeList {
+    std::map<uint64_t, size_t> _free_areas;
```

Comment:
Consider adding a comment on what is the key and value. Also applies for __free_sizes_ under.

Reply:
Done.